optuna/integration/_lightgbm_tuner/sklearn.py

```diff
@@ -1,23 +1,18 @@
+from typing import Any  # NOQA
+from typing import Dict  # NOQA
+from typing import List  # NOQA
 import warnings
 
 import lightgbm as lgb
 
-from optuna import type_checking
-
-if type_checking.TYPE_CHECKING:
-    from typing import Any  # NOQA
-    from typing import Dict  # NOQA
-    from typing import List  # NOQA
-
 
 class LGBMModel(lgb.LGBMModel):
     """Proxy of lightgbm.LGBMModel.
 
     See: `pydoc lightgbm.LGBMModel`
     """
 
-    def __init__(self, *args, **kwargs):
-        # type: (List[Any], Dict[str, Any]) -> None
+    def __init__(self, *args: List[Any], **kwargs: Dict[str, Any]) -> None:
```

Comment:
Should the type of `args` be `Tuple` instead of `List`?

```python
def f(*args, **kwargs):
    print(type(args), type(kwargs))


f(1, a="a")
```

outputs:

```
<class 'tuple'> <class 'dict'>
```

Reply:
I agree with @harupy. How about changing the type annotation from `List[Any]` to `Tuple[Any]`?

Reply:
In my understanding, type hints of `*args` can be a special type like `str` or `List`. After receiving them, the type of `args` is dedicated as `Tuple`, isn't it?
https://www.python.org/dev/peps/pep-0484/#arbitrary-argument-lists-and-default-argument-values

Reply:
looks like we should use `Any` for both `args` and `kwargs`.

Reply:
Thanks for providing useful information. According to the reference, 
> In the body of function foo, the type of variable args is deduced as Tuple[str, ...] and the type of variable kwds is Dict[str, int].

I'm not sure how `mypy` checks the type hinting in the function signature (in the signature or in the function body), but the current one does not occur any `mypy` errors. I think `List[Any]` in the function signature is OK because it is easy to understand and no error occurs in `mypy`'s check. What do you think?

Reply:
@HideakiImamura Tested using `mypy` and vscode:

![mypy-check](https://user-images.githubusercontent.com/17039389/90718688-84b80380-e2ed-11ea-9d11-48110fb77bdc.gif)

Reply:
If we use `*args: List[Any]`, `LGBModel` only accepts `List[Any]` as positional arguments.

Reply:
Since we're only forwarding the arguments to `lightgbm` without any assumptions whatsoever, `Any` for both `*args` and `**kwargs` as suggested by @harupy sounds robust and rather reasonable to me.

Reply:
Thanks for the investigation. Actually, I agree that we should use `Any` instead of `List[Any]` here.

Reply:
Thank you! Really helpful. So, as a scope of this PR, is it OK to use `Any` for both `args` and `kwargs`?